tests/components/cloud/test_http_api.py

```diff
@@ -1,87 +1,114 @@
+        },
+    ],
+    "preferred_item": "12345",
+}
 
-@pytest.fixture(name="mock_cloud_login")
-def mock_cloud_login_fixture(hass, setup_api):
-    """Mock cloud is logged in."""
-    hass.data[DOMAIN].id_token = jwt.encode(
+PIPELINE_DATA_OTHER = {
+    "items": [
         {
-            "email": "hello@home-assistant.io",
-            "custom:sub-exp": "2018-01-03",
-            "cognito:username": "abcdefghjkl",
+            "conversation_engine": "other",
+            "conversation_language": "language_1",
+            "id": "12345",
+            "language": "language_1",
+            "name": "Home Assistant",
+            "stt_engine": "stt.other",
+            "stt_language": "language_1",
+            "tts_engine": "other",
+            "tts_language": "language_1",
+            "tts_voice": "Arnold Schwarzenegger",
+            "wake_word_entity": None,
+            "wake_word_id": None,
         },
-        "test",
-    )
+    ],
+    "preferred_item": "12345",
+}
+
+SUBSCRIPTION_INFO_URL = "https://api-test.hass.io/payments/subscription_info"
 
 
-@pytest.fixture(autouse=True, name="setup_api")
-def setup_api_fixture(hass, aioclient_mock):
-    """Initialize HTTP API."""
-    hass.loop.run_until_complete(
-        mock_cloud(
-            hass,
-            {
+@pytest.fixture(name="setup_cloud")
+async def setup_cloud_fixture(hass: HomeAssistant, cloud: MagicMock) -> None:
+    """Fixture that sets up cloud."""
+    assert await async_setup_component(hass, "homeassistant", {})
+    assert await async_setup_component(
+        hass,
+        DOMAIN,
+        {
+            DOMAIN: {
                 "mode": "development",
                 "cognito_client_id": "cognito_client_id",
                 "user_pool_id": "user_pool_id",
                 "region": "region",
+                "alexa_server": "alexa-api.nabucasa.com",
                 "relayer_server": "relayer",
                 "accounts_server": "api-test.hass.io",
                 "google_actions": {"filter": {"include_domains": "light"}},
                 "alexa": {
                     "filter": {"include_entities": ["light.kitchen", "switch.ac"]}
                 },
             },
-        )
+        },
     )
-    return mock_cloud_prefs(hass)
+    await hass.async_block_till_done()
+    on_start_callback = cloud.register_on_start.call_args[0][0]
+    await on_start_callback()
 
 
 @pytest.fixture(name="cloud_client")
-def cloud_client_fixture(hass, hass_client):
+async def cloud_client_fixture(
+    hass: HomeAssistant, hass_client: ClientSessionGenerator
+) -> TestClient:
     """Fixture that can fetch from the cloud client."""
-    with patch("hass_nabucasa.Cloud._write_user_info"):
-        yield hass.loop.run_until_complete(hass_client())
-
-
-@pytest.fixture(name="mock_cognito")
-def mock_cognito_fixture():
-    """Mock warrant."""
-    with patch("hass_nabucasa.auth.CognitoAuth._cognito") as mock_cog:
-        yield mock_cog()
+    return await hass_client()
```

Comment:
I'd suggest to remove this fixture since it doesn't seem to be needed anymore, and just update tests to use the `hass_client` fixture.

Reply:
I just added it to keep the diff smaller.

Reply:
I'll do this in a follow up PR next.